admission-webhook/integration_tests/integration_test.go

```diff
@@ -273,6 +288,68 @@ func TestPossibleToUpdatePodWithExistingGMSASettings(t *testing.T) {
 	assert.True(t, success)
 }
 
+func TestDeployV1Alpha1CredSpecGetAllVersions(t *testing.T) {
+	testName := "deploy-v1alpha1-credspec-get-all-versions"
+	credSpecTemplates := []string{"credspec-0", "credspec-1"}
```

Comment:
it looks like some of the other tests also use this `credspec-1` which hasn't been updated to the latest version. does this mean they are running the tests on the alpha version? Should we upgrade them all and add a new spec version called `credspec-alpha` to be clear that is what is being tested?

Reply:
I set the storage version to v1 so unless any of the tests are explicitly asking for v1alpha1 they should be getting v1 specs

Reply:
I don't know if understand the full implications of using the v1alpha across the tests but seems like it would be valuable to have a test that deploys it and uses it. For our own sanity in the future should we rename the credspec-1 to credspec-v1alpha1 so we know this was intentional?  

Do we need a credspec-1 that is v1 so test the scenarios that credspec-1 was testing?

Reply:
> Do we need a credspec-1 that is v1 so test the scenarios that credspec-1 was testing?

I don't think we need a credspec-1 that is v1.
The different credspec templates that are checked in specify different group managed service accounts that can be used for different containers.
The existing tests all validate the contents of the credspec and to make sure the mutating webhook properly applies (or rejects) populating the windowsSecurityContext.credentialSpec fields in a manner that is agnostic to the the CRD version.

See:
https://github.com/kubernetes-sigs/windows-gmsa/blob/85e30b7cb547fd7ac1a408cec9bb8313bee221c0/admission-webhook/integration_tests/integration_test.go#L20-L26

All existing tests that referenced credspec-1 will deploy and use that template.
The PR sets the storage version for the CRD to v1 and updates the webhook to query for v1 GMSACredentialSpec objects but still allows for people to retried v1alpha1 objects (serve = true).

The additional tests that I added ensure that the contents of GMSACredentialSpec objects are exactly the same regardless of if v1alpha1 objects or v1 objects were requested (for scenarios where either v1alpha1 or v1 objects were written).

Reply:
> The existing tests all validate the contents of the credspec and to make sure the mutating webhook properly applies (or rejects) populating the windowsSecurityContext.credentialSpec fields in a manner that is agnostic to the the CRD version.

sounds good.  Thanks for helping clarify.

Leaving a comment in the one template as too why its not V1 for the tests might help future us but will leave it up to you